text/0000-export-to-oci.md

```diff
@@ -0,0 +1,410 @@
+| `<run-image>`	| Run image reference |
+
+<!-- | `<cache-image>`	| Reference to a cache image in an OCI image registry | -->
+
+Let's suppose a directory exits and it has the following structure:
+
+```=shell
+oci
+├── cnb
+│   └── my-stack-run
+│       ├── blobs
+│       │   └── sha256
+│       │       └── 1bcd..x
+│       ├── index.json
+│       └── oci-layout
+├── foo
+│   └── my-cache
+│       ├── blobs
+│       │   └── sha256
+│       │       └── 1f591..a
+│       ├── index.json
+│       └── oci-layout
+└── bar
+    └── my-previous-app
+        ├── blobs
+        │   └── sha256
+        │       └── 1efg..w
+        ├── index.json
+        └── oci-layout
+```
+
+For each case, I will present two ways of enabling the new capability:
+
+- Using an environment Variable
+- Using a new `oci` flag
+
+In any case the expected output is the same.
+
+#### Analyze phase
+
+##### Analyzing run-image
+
+```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/analyzer -run-image cnb/my-stack-run:bionic my-app-image
+
+# OR
+
+> /cnb/lifecycle/analyzer -oci -run-image cnb/my-stack-run:bionic my-app-image
+
+# expected output
+# analyzed.toml file with the usual `run-image.reference`
+
+```
+
+##### Analyzing previous-image
+
+ ```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/analyzer -run-image cnb/my-stack-run:bionic -previous-image bar/my-previous-app my-app-image
+
+# OR
+
+> /cnb/lifecycle/analyzer -oci -run-image cnb/my-stack-run:bionic-previous-image bar/my-previous-app my-app-image
+
+# expected output
+# analyzed.toml file with the usual `run-image.reference` and `previos-image.reference`
+```
+<!--
+##### Analyzing using cache-image
+
+```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/analyzer -run-image cnb/my-stack-run:bionic -cache-image foo/my-cache my-app-image
+
+# OR
+
+> /cnb/lifecycle/analyzer -oci -run-image cnb/my-stack-run:bionic -cache-image foo/my-cache my-app-image
+
+# expected output
+# analyzed.toml file with the usual `run-image.reference`
+
+```
+
+-->
+
+##### Analyzing run-image not saved on disk
+
+```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/analyzer -run-image cnb/bad-run-image my-app-image
+
+# OR
+
+> /cnb/lifecycle/analyzer -oci -run-image cnb/bad-run-image my-app-image
+
+# expected output
+
+ERROR: the run-image could not be found at path: /oci/cnb/bad-run-image
+```
+
+##### Analyzing without run-image argument
+
+```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/analyzer my-app-image
+
+# OR
+
+> /cnb/lifecycle/analyzer -oci my-app-image
+
+# expected output
+
+ERROR: -run-image is required when OCI feature is enabled
+```
+
+#### Export phase
+
+##### Export to OCI
+
+
+```=shell
+> export CNB_USE_OCI=true
+> /cnb/lifecycle/exporter -run-image cnb/my-stack-run:bionic my-app-image
+
+# OR
+
+>  /cnb/lifecycle/exporter -oci -run-image cnb/my-stack-run:bionic my-app-image
+
+# expected output
+# the store folder will be updated with the application image as follows
```

Comment:
Did we talk about what level of compression (if any) is applied to the layers? Can this be configurable?